services/files/src/storage.ts

```diff
@@ -0,0 +1,71 @@
+import { GetSignedUrlConfig, Storage } from "@google-cloud/storage";
+import config from "@api/config";
+import path from "path";
+import { ObjectId } from "mongodb";
+
+import { FileModel } from "./models/file";
+
+const storage = new Storage();
+
+const bucket = storage.bucket(config.services.FILES.storageBucket || "");
+
+export const uploadFile = async (file: Express.Multer.File, userId: string): Promise<string> => {
+  const { originalname, buffer } = file;
+
+  const googleFileName = `${path.parse(originalname).name}_${Date.now()}`;
+  const blob = bucket.file(googleFileName);
+
+  const blobStream = blob.createWriteStream({
+    resumable: false,
+  });
+
+  const { id } = await FileModel.create({
+    fileName: path.parse(originalname).name,
+    mimeType: file.mimetype,
+    userId,
+    storageId: googleFileName,
+  });
+
+  blobStream
+    .on("finish", async () => {
+      await blob.makePublic();
+      await blob.setMetadata({
+        contentType: file.mimetype,
+      });
+    })
+    .on("error", err => {
+      throw new Error(err.message);
```

Comment:
this one i think u could keep a normal error bc the error could be caused by other things